Source/WebCore/animation/ScrollTimeline.cpp

```diff
@@ -166,11 +166,22 @@ ScrollTimeline::Data ScrollTimeline::computeScrollTimelineData() const
     // https://drafts.csswg.org/scroll-animations-1/#scroll-timeline-progress
     // Progress (the current time) for a scroll progress timeline is calculated as:
     // scroll offset ÷ (scrollable overflow size − scroll container size)
+    // TODO: progress calculation now (scrollOffset - rangeStart) / (rangeEnd - rangeStart)
 
     float maxScrollOffset = axis() == ScrollAxis::Block ? sourceScrollableArea->maximumScrollOffset().y() : sourceScrollableArea->maximumScrollOffset().x();
     float scrollOffset = axis() == ScrollAxis::Block ? sourceScrollableArea->scrollOffset().y() : sourceScrollableArea->scrollOffset().x();
 
-    return { maxScrollOffset, scrollOffset };
+    auto computeRangeEnd = [&](SingleTimelineRange range) -> float {
+        ASSERT(range.range == NamedTimelineRange::Normal || range.range == NamedTimelineRange::Omitted);
+        return maxScrollOffset - floatValueForLength(range.offset, maxScrollOffset);
+    };
+
+    auto computeRangeStart = [&](SingleTimelineRange range) -> float {
+        ASSERT(range.range == NamedTimelineRange::Normal || range.range == NamedTimelineRange::Omitted);
+        return floatValueForLength(range.offset, maxScrollOffset);
+    };
```

Comment:
Not sure we gain anything using lambdas here since these functions are only used one so using a parameter seems pointless.

Reply:
Right yeah lambdas aren't necessary for this one.